### DISCUSSION TITLE
Who is responsible for Token Economics of ERC20 tokens?

### DISCUSSION THREAD

**Discussion by alymurtazamemon** [Cyfrin/foundry-full-course-cu#87] [Q&A] [answered]
@PatrickAlphaC is it the developers' work to handle the token economics of token or do they have a separate team/people for it?

If it is the developer's job then I am not seeing anything around it in your tutorials like what should be the total supply of tokens? what should be the initial price? etc.

And as far as I know, to give a real value to our tokens we need to create a pair of them with tokens such as BTC, ETC, USDC, etc and deposit them in the liquidity pool. So my question is how the first token BTC got its real value because at that time there were no other tokens to pair with.

**Comment by devblixt** [comment]
You don't necessarily need liquidity pools for tokens to 'have value'. Think of NFTs- they get value because of the community or art they represent. Similarly, Bitcoin initially had its own enthusiasts plus an inbuilt notion of scarcity, thus satisfying supply-demand economics. Even without liquidity pools, you can have limit-order/order-book marketplaces or even P2P ones. 

Coming to tokenomics, it mainly depends on what kind of project you are building. Let's say you're building an omnichain messaging system like Layerzero or an Type-1-based L2 like Taiko. These have a high chance of gaining widespread adoption, so your valuations become high. But if you're building a very niche app like let's say, an app that allows communities to migrate from one NFT standard to another. Such a project would have a smaller valuation. 

Total supply, token distribution, initial price, etc are largely dependent on market trends and the team which builds a project. Who decides these? Well, it is more of a team choice. I've seen chief developers themselves deciding tokenomics, and I've seen teams hiring people to decide them as well.

**Comment by PatrickAlphaC** [comment, marked as answer]
Great question. The answer is "it depends". I don't do a lot of token tutorials because it gets very hairy depending on where you're located, and to be honest, for the most part, most protocols don't need a token.

I mention in the governance lecture, that I don't like ERC20 based governance, I think it's a flawed design.

I'd like to see new experiments:

- DACs instead of DAOs (decentralized corporations, where leaders are elected)
- Privacy-preserving reputation-based voting
- skin in the game

You ask an amazing question about "How did BTC get it's value in the first place?" and I've been meaning to make a "what is money" video for some time.

To roughly answer you here, let me ask you this question "why does gold have value?". My answer to that is "gold has value because other people think it has value". Gold has value, because people think that other people will want it, and in that regard it's a "confidence asset" meaning it's backed by vIbEs. I'm grossly glossing over this, but you get the picture.

IMO, BTC has this same narrative, it has value because others think it has value. Now you might say "OMG wtf???" and I'd say that's a good response, however, we can ask this about any asset. For example, what is USD backed by? 

Well, it _used_ to be backed by gold. You could exchange 1 USD for it's equivalent in gold at any time prior to 1971. But in 1971, the USA made the choice to [no longer have the dollar backed by gold](https://www.investopedia.com/ask/answers/09/gold-standard.asp) and in doing so, essentially made the dollar backed by... them. The dollar now has value because other people think it has value. 

This is where ETH & Collateralized coins come in. They think that having a coin backed by vibes is silly.

- ETH value (at it's core) comes from the ability to create smart contracts
- Stablecoins gain value from their collateral
- Oil has value because you can use it to heat your house

etc

**Reply by alymurtazamemon** [reply]
@PatrickAlphaC Yes like the earliest form of trade involved bartering, which is the exchange of goods or services directly without using money. People started exchanging $$$ for BTC in the early days.

Ok so now this is clear but now when we create an ERC20 token, how as a developer we can decide the price of it? and total supply, etc. Is there any resource you know where we can learn it?

Let's say I want my token to be back with ETH now.

**Reply by PatrickAlphaC** [reply]
If you want to decide the price of a token, there are a few ways to do this:

1. Have a set exchange rate (like what we do in the stablecoin project)
2. Let the market decide

"Choosing" a price can be hard, if anyone could just "do" this, I could say "I have 1,000,000 PATRICK coins and each coin is worth $1, so therefore I'm a millionaire!"

The market typically decides the price of an asset. 

However, in our stablecoin project, we make the price of 1 DSC = $1 by using collateral, be sure to check out lesson 12 to learn more, and then let me know!

**Reply by rohit-bhalerao-7** [reply]
Please make a video on "What is money" as you said, eagerly waiting for it!

**Reply by PatrickAlphaC** [reply]
I need to move it higher on my TODO list ahah. Great call.

**Reply by sharonphiliplima** [reply]
We are waiting for the "What is Money" video, ser Patrick.

**Reply by PatrickAlphaC** [reply]
Good shout.

**Comment by TarasBluzz** [comment]
Guys, where I can find information about token supply?

**Comment by ghost** [comment]
@alymurtazamemon @TarasBluzz, this repo might help you: https://github.com/jpantunes/awesome-cryptoeconomics
